From 43f21d17de82755344f60fe7a026bc3566a89dfb Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Mon, 18 Mar 2024 11:13:56 +0800
Subject: [PATCH 008/116] wifi: mt76: mt7992: adjust beamform mcu cmd
 configuration for mt7992

Adjust the correct beamform mcu cmd configuration for mt7992 chipsets.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
CR-Id: WCNCR00240772
Change-Id: I2230c3a43d735ab7db114299a005b9cc1e89de41
---
 mt7996/mcu.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index f23801901..7928571e0 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -3920,8 +3920,9 @@ int mt7996_mcu_set_txbf(struct mt7996_dev *dev, u8 action)
 
 		tlv = mt7996_mcu_add_uni_tlv(skb, action, sizeof(*req_mod_en));
 		req_mod_en = (struct bf_mod_en_ctrl *)tlv;
-		req_mod_en->bf_num = 3;
-		req_mod_en->bf_bitmap = GENMASK(2, 0);
+		req_mod_en->bf_num = mt7996_band_valid(dev, MT_BAND2) ? 3 : 2;
+		req_mod_en->bf_bitmap = mt7996_band_valid(dev, MT_BAND2) ?
+					GENMASK(2, 0) : GENMASK(1, 0);
 		break;
 	}
 	default:
-- 
2.39.2

